modules/caddyhttp/reverseproxy/httptransport.go

```diff
@@ -369,6 +380,13 @@ type TLSConfig struct {
 	//  - "once": allows a remote server to request renegotiation once per connection.
 	//  - "freely": allows a remote server to repeatedly request renegotiation.
 	Renegotiation string `json:"renegotiation,omitempty"`
+
+	// Skip TLS ports specifies a list of upstream ports on which TLS should not be
+	// attempted even if it is configured. Handy when using dynamic upstreams that
+	// return HTTP and HTTPS endpoints too.
+	// When specified, TLS will automatically be configured on the transport.
+	// The value can be a list of any valid tcp port numbers, default empty.
+	SkipTLSPorts []string `json:"skip_tls_ports,omitempty"`
```

Comment:
```suggestion
	ExceptPorts []string `json:"except_ports,omitempty"`
```

What about this? (Current name suggests that we're skipping TLS ports, but the opposite is true.) Maybe also reword it in the documentation comment.

(And "TLS" might be redundant since we're in a TLSConfig struct.

Reply:
Another thought is that it's more natural to express port numbers as integers. I know that Go uses strings, but I wonder if there's any reason a port would need to be a string...

Reply:
Thanks @mholt, your naming is much better than mine.

I agree that port 80 should be default because of it's historical status as plain HTTP port, but adding it has a slight chance of breaking current installations because it autoconfigures TLS, leaving it empty doesn't break or affect anything.

I chose the value simply because it fits in nicely with how GO handles port numbers and it looks clean this way. Building json configs will look awkward, given that you have to specify numbers as strings, but the code to handle it would look much more cumbersome.